test/unit-test/ota_utest.c

```diff
@@ -1643,6 +1674,27 @@ void test_OTA_ReceiveFileBlockCompleteDynamicBufferHttp()
     test_OTA_ReceiveFileBlockCompleteHttp();
 }
 
+/* Test that extractAndStoreArray fails if device does not have sufficient
+ * memory to allocate the string/array. */
+void test_OTA_ExtractArrayMemAllocFails()
+{
+    otaInterfaces.os.mem.malloc = mockMallocAlwaysFail;
+
+    memset( &pOtaAppBuffer, 0, sizeof( pOtaAppBuffer ) );
+    otaGoToState( OtaAgentStateWaitingForFileBlock );
+    TEST_ASSERT_EQUAL( OtaAgentStateWaitingForJob, OTA_GetState() );
+}
+
+/* Test that extractAndStoreArray fails if user buffer is insufficient
+ * to allocate the string/array (here streamname). */
+void test_OTA_ExtractArrayInsufficientBuffer()
+{
+    pOtaAppBuffer.streamNameSize = 5;
+
+    otaGoToState( OtaAgentStateWaitingForFileBlock );
+    TEST_ASSERT_EQUAL( OtaAgentStateWaitingForJob, OTA_GetState() );
```

Comment:
You may want to consider adding a brief description here as to why this is the expected behavior. It's not very clear why going to the `OtaAgentStateWaitingForFileBlock ` state means that we should end up in the `OtaAgentStateWaitingForJob` state. The step in the process of getting to the waiting for file block state and then ending up in the waiting for job state that fails is not clear to me.

Reply:
Added description in https://github.com/aws/ota-for-aws-iot-embedded-sdk/pull/171/commits/3f638123d8d8f787b75bdf3e2663b0ee10f9cad5